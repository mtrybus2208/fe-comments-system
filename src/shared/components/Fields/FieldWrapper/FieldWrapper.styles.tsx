import styled, { css } from 'styled-components';

export const Container = styled.div`
  position: relative;
  display: grid;
  width: 100%;
  align-items: center;
  grid-template-areas:
    'label label'
    'input input'
    'error error';
  grid-template-rows: 1fr;
  grid-template-columns: 1fr;
`;

export const Content = styled.div`
  grid-area: input;
`;

export const FieldWrapper = styled.div`
  display: block;
`;
