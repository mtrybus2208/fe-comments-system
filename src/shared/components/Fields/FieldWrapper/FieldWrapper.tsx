import React from "react";

import FieldLabel from "../FieldLabel/FieldLabel";
import ErrorMsg from "./components/ErrorMsg/ErrorMsg";
import * as S from "./FieldWrapper.styles";
import { selectValidationType } from "../../../helpers/forms/selectValidationType";
import { InputValidationTypes } from "../../../../types/shared/forms.types";

export interface FieldWrapperProps {
  children: JSX.Element[] | JSX.Element;
  label: string;
  name: string;
  error?: string;
  touched?: boolean;
  validationType?: InputValidationTypes;
}

const FieldWrapper: React.FunctionComponent<FieldWrapperProps> = ({
  children,
  name,
  label,
  error,
  touched,
  validationType,
}) => {
  return (
    <S.Container>
      <FieldLabel htmlFor={name} name={label} />
      <S.Content>{children}</S.Content>
      <ErrorMsg
        msg={error}
        show={!!selectValidationType({ validationType, touched, error })}
      />
    </S.Container>
  );
};

export default FieldWrapper;
