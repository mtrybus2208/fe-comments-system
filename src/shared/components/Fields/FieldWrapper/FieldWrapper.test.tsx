import React from 'react';
import { shallow } from 'enzyme';

import FieldWrapper from './FieldWrapper';

describe('FieldWrapper', () => {
  it('should render layout', () => {
    const props = {
      label: 'foo',
      name: 'bar',
    };
    const wrapper = shallow(
      <FieldWrapper {...props}>
        <span>42</span>
      </FieldWrapper>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
