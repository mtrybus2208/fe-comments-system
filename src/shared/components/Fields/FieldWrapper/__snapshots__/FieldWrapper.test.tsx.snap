// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldWrapper should render layout 1`] = `ShallowWrapper {}`;
